### PR TITLE
Update Xsemantics to Oxygen and Xtext 2.13

### PR DESCRIPTION
**Pull request by qtran-n4**

### CODE REVIEW

File: devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup [hunk 1]
```diff
@@ -189,7 +189,7 @@
       <requirement
           name="org.eclipse.b3.aggregator.editor.feature.feature.group"/>
       <repository
-          url="http://download.eclipse.org/modeling/emft/b3/updates-4.4"/>
+          url="http://download.eclipse.org/cbi/updates/aggregator/ide/4.4/?d"/>
```
Comment (LorenzoBettini) [inline devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup]: I think that the "?d" is wrong: it's useful only for browsing the contents via web but it does not work with p2. I also think that for Oxygen the "4.4" should be increased... In any case, I'm not using the aggregator with Xsemantics anymore, so we could also remove such an installed feature from the Oomph setup

File: devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup [hunk 2]
```diff
@@ -525,22 +525,22 @@
       </query>
     </setupTask>
   </setupTask>
-  <stream name="master">
+  <stream name="GH-370-2">
```
Comment (LorenzoBettini) [inline devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup]: this should not be changed! Please leave "master".

File: devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup [hunk 3]
```diff
@@ -525,22 +525,22 @@
...
-        value="http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.12.0"/>
+        value="http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.13.0"/>
     <setupTask
         xsi:type="setup.p2:P2Task"
         id="p2.org.eclipse.xtext.2.10"
         label="Xtext Releases">
       <requirement
           name="org.eclipse.xtext.sdk.feature.group"
-          versionRange="[2.12.0,2.13.0)"/>
+          versionRange="[2.13.0,2.13.1)"/>
```
Comment (LorenzoBettini) [inline devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup]: too strict: it should be [2.13.0,2.14.0)

File: devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup [hunk 4]
```diff
@@ -560,14 +560,14 @@
...
-        value="http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.11.0"/>
+        value="http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.13.0"/>
     <setupTask
         xsi:type="setup.p2:P2Task"
         id="p2.org.eclipse.xtext.nightly"
         label="Xtext Nightly">
       <requirement
           name="org.eclipse.xtext.sdk.feature.group"
-          versionRange="[2.11.0,2.12.0)"/>
+          versionRange="[2.13.0,2.13.1)"/>
```
Comment (LorenzoBettini) [inline devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup]: as above

**Comment by LorenzoBettini** [comment]
An additional remark: you haven't updated the target platform so that it uses Oxygen and Xtext 2.13 during the Maven build (that's why the Travis build fails): you need to create a new target file (starting from an existing one, for example) and make sure you refer to that new target file name in the pom.

File: devtools/org.eclipse.xsemantics.targetplatform/oxygen-xtext-2.13.target [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+<unit id="de.itemis.xtext.antlr.feature.feature.group" version="2.1.1.v201405091103"/>
+<repository location="http://download.itemis.com/updates/releases/2.1.1/"/>
+</location>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.eclipse.sdk.feature.group" version="0.0.0"/>
+<unit id="org.eclipse.emf.sdk.feature.group" version="0.0.0"/>
+<unit id="org.eclipse.xpand" version="0.0.0"/>
+<unit id="org.eclipse.xtend" version="0.0.0"/>
+<unit id="org.eclipse.xtend.typesystem.emf" version="0.0.0"/>
+<repository location="http://download.eclipse.org/releases/neon/"/>
```
Comment (LorenzoBettini) [inline devtools/org.eclipse.xsemantics.targetplatform/oxygen-xtext-2.13.target]: You're still using neon, not oxygen

File: devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup (hunk 2 above)
Comment (LorenzoBettini) [reply devtools/org.eclipse.xsemantics.workspace/Xsemantics.setup]: Please, leave "master"

File: releng/org.eclipse.xsemantics.parent/pom.xml [hunk 6]
```diff
@@ -120,7 +120,7 @@ Contributors:
 		<swtbot.tests.only>false</swtbot.tests.only>
 		<swtbot.tests.skip>false</swtbot.tests.skip>
 		
-		<target-file-name>neon-xtext-2.12</target-file-name>
+		<target-file-name>oxygen-xtext-2.13.target</target-file-name>
```
Comment (LorenzoBettini) [inline releng/org.eclipse.xsemantics.parent/pom.xml]: You must specify ONLY the name of the .target file, without ".target"

**Comment by qtran-n4** [comment]
@LorenzoBettini Thanks a lot for your quick review Lorenzo! I am leaving the stream to GH-370-2 so I can oomph that branch for testing. I will reverse it to master before merging to master

File: releng/org.eclipse.xsemantics.parent/pom.xml [hunk 7]
```diff
@@ -63,7 +63,7 @@ Contributors:
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<xtext.version>2.13.0</xtext.version>
 		<xtextVersion>${xtext.version}</xtextVersion>
-		<mwe2Version>2.9.0.201605261059</mwe2Version>
+		<mwe2Version>2.9.1</mwe2Version>
```
Comment (LorenzoBettini) [inline releng/org.eclipse.xsemantics.parent/pom.xml]: You must specify the complete version, I think it is 2.9.1.201705291010 (see the Travis log)

**Comment by LorenzoBettini** [comment]
There's still a failure in the Travis build: the .xsemantics files in the FJ cannot be compiled with the xtext-maven-plugin; I've seen this problem in the past and I have no clue on how to solve this, so I'd suggest you to comment out the xtext-maven-plugin section in the pom of the fj project.

**Comment by qtran-n4** [comment]
@LorenzoBettini The same problem occurs for expression and lambda project.

**Comment by cdietrich** [comment]
What kind of error do you get?

**Comment by qtran-n4** [comment]
@cdietrich For instance while compiling `expressions-cached.xsemantics`

```
judgments {
	override vartype ||- Variable variable : output Type
		cached
		error "cannot type " + stringRep(variable)
		source variable
		feature ExpressionsPackage::eINSTANCE.variable_Expression
}
```

We receive validation error messages saying that for instance, `Type` is not valid..

**Comment by cdietrich** [comment]
hmmm i think that should be investigated

**Comment by LorenzoBettini** [comment]
@cdietrich if I remember correctly, the example of fj gives problem if the type system that extends another fj type system is processed first then you have compilation errors, and I never had time to investigate it further. The problem with the lambda example is a novelty though :( the build log https://travis-ci.org/eclipse/xsemantics/builds/341821585 shows a ClassCastException 

java.lang.ClassCastException: org.eclipse.jdt.internal.compiler.batch.ClasspathJar cannot be cast to java.lang.String
	at org.eclipse.jdt.internal.compiler.apt.util.EclipseFileManager.getDefaultBootclasspath(EclipseFileManager.java:257)

that I've seen in other projects, but in the other projects it never had consequences... while in this case it might be the cause...

**Comment by qtran-n4** [comment]
@LorenzoBettini:

- The Travis build is now green.
- Oomphing my branch (GH-370-2) works fine.
- I run the test in Xsemantics project: dsl.ui.tests are all green.  dsl.tests has 4 errors but no failures (I think the errors are intentional)
screen shot 2018-02-15 at 13 52 57

-  Keep in mind that I have commented out `xtext-maven-plugin` in fj, expression and lambda projects.

**Comment by qtran-n4** [comment]
@LorenzoBettini Also Lorenzo we should increase the Xsemantics version as well? The current version of Xsemantics is 1.12.1. If yes, should we increase the major or minor version?

**Comment by LorenzoBettini** [comment]
No, the errors are not intentional: there must be error in tests, indeed those tests are not failing in Travis, are they?

Version increment must be done on another commit, after this PR has been merged.

**Comment by qtran-n4** [comment]
@LorenzoBettini : I have reverted the Xsemantics back to 1.12.1 and fixed the error tests (see diff). Please let me know if there is any other problem :-)

**Comment by LorenzoBettini** [comment]
I think we're fine now, please remove the branch from the oomph setup.

By the way, you don't need to specify a different branch in the oomph setup: the branch name is only used the first time for creating the workspace, but you can still switch to any branch you like after that.

**Comment by qtran-n4** [comment]
@LorenzoBettini  Thanks for the hint! I have reverted the stream to master. If everything's fine, please approve the PR and merge it to master. As a next step, I will upgrade N4JS IDE to Oxygen and Xtext 2.13.

**Pull request merged**